Monika After Story/game/script-stories.rpy

```diff
@@ -11,39 +11,121 @@
 
 # dict of tples containing the stories event data
 default persistent._mas_story_database = dict()
-default mas_can_unlock_story = False
-default mas_can_unlock_scary_story = False
+
+#Debug var to test scares
 default mas_full_scares = False
+
 # dict storing the last date we saw a new story of normal and scary type
-default persistent._mas_last_seen_new_story = {"normal":None,"scary":None}
+default persistent._mas_last_seen_new_story = {"normal": None, "scary": None}
 
+#Flag to mark if the setup for scary stories has been done
+define mas_scary_story_setup_done = False
 
 # store containing stories-related things
 init -1 python in mas_stories:
     import store
+    import datetime
+
+    UNLOCK_NEW = "unlock_new"
 
     # TYPES:
-    TYPE_SCARY = 0
+    TYPE_NORMAL = "normal"
+    TYPE_SCARY = "scary"
 
     # pane constant
     STORY_RETURN = "Nevermind"
     story_database = dict()
 
+    #Time between story unlocks of the same type (in hours). Changes over sessions, but also changes after the next story unlocks
+    TIME_BETWEEN_UNLOCKS = renpy.random.randint(20, 28)
+
+    #TODO: Build functions to 'register' story types. This will add all related info to the map
+    #saving the manual additions to each part
+
+
+    #Story which starts unlocked for a specific type
+    FIRST_STORY_EVL_MAP = {
+        TYPE_SCARY: "mas_scary_story_hunter",
+        TYPE_NORMAL: "mas_story_tyrant"
+    }
+
+    #Override maps to have special conditionals for specific story types
+    NEW_STORY_CONDITIONAL_OVERRIDE = {
+        TYPE_SCARY: "mas_isO31() or mas_stories.check_can_unlock_new_story(mas_stories.TYPE_SCARY)",
+    }
+
+    def check_can_unlock_new_story(story_type=TYPE_NORMAL):
+        """
+        Checks if it has been at least one day since we've seen the last story or the initial story
+
+        IN:
+            story_type - story type to check if we can unlock a new one
+                (Default: TYPE_NORMAL)
+        """
+        global TIME_BETWEEN_UNLOCKS
+
+        new_story_ls = store.persistent._mas_last_seen_new_story.get(story_type, None)
+
+        #Get the first story of this type
+        first_story = FIRST_STORY_EVL_MAP.get(story_type, None)
+
+        #If this doesn't have an initial, no go
+        if not first_story:
+            return False
+
+        can_show_new_story = ((
+                store.seen_event(first_story) if new_story_ls is None
+                else store.mas_timePastSince(new_story_ls, datetime.timedelta(hours=TIME_BETWEEN_UNLOCKS))

```

Comment:
`persistent._mas_last_seen_new_story` should hold `datetime` objects. Otherwise `mas_timePastSince` combines it with `time(0, 0)` (midnight) and `TIME_BETWEEN_UNLOCKS` becomes irrelevant.